compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -75,11 +75,11 @@ enum LetSource {
     WhileLet,
 }
 
-struct MatchVisitor<'thir, 'p, 'tcx> {
+struct MatchVisitor<'p, 'tcx> {
```

Comment:
We're very lucky that `'thir` is covariant here -- that means that we can shorten it to `'p`, which is the lifetime of our pattern interner.